src/data/bucket/fill_extrusion_bucket.js

```diff
@@ -565,9 +573,13 @@ register(PartMetadata, 'PartMetadata');
 
 export default FillExtrusionBucket;
 
-function isBoundaryEdge(p1, p2, bounds) {
-    return (p1.x === p2.x && (p1.x < bounds[0].x || p1.x > bounds[1].x)) ||
-           (p1.y === p2.y && (p1.y < bounds[0].y || p1.y > bounds[1].y));
+// Edges that are outside bounds are defined in tile across the border.
+// Rendering them twice often results with Z-fighting.
+function isEdgeOutsideBounds(p1, p2, bounds) {
+    return (p1.x < bounds[0].x && p2.x < bounds[0].x) ||
+           (p1.x > bounds[1].x && p2.x > bounds[1].x) ||
+           (p1.y < bounds[0].y && p2.y < bounds[0].y) ||
+           (p1.y > bounds[1].y && p2.y > bounds[1].y);
 }
```

Comment:
This function is used to cull edges outside of the provided boundaries which doesn't have to be tile boundaries. For example fill extrusion reprojection for the globe view would otherwise generate internal edges to the boundaries of the clip regions. These kind of edges are always axis aligned. (This comment is mostly about the comment added to the function :) )